src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContext.cs

```diff
@@ -7,78 +7,49 @@
         {
-            Requires.NotNullOrEmpty(configuredProjectContextsByTargetFramework, nameof(configuredProjectContextsByTargetFramework));
+            Requires.Argument(!targetFrameworks.IsDefaultOrEmpty, nameof(targetFrameworks), "Must contain at least one item.");
             Requires.NotNullOrEmpty(configuredProjectsByTargetFramework, nameof(configuredProjectsByTargetFramework));
             Requires.NotNull(activeTargetFramework, nameof(activeTargetFramework));
-            Requires.Argument(configuredProjectContextsByTargetFramework.ContainsKey(activeTargetFramework),
-                              nameof(configuredProjectContextsByTargetFramework), "Missing ICrossTargetProjectContext for activeTargetFramework");
+            Requires.Argument(targetFrameworks.Contains(activeTargetFramework), nameof(targetFrameworks), "Must contain 'activeTargetFramework'.");
 
             IsCrossTargeting = isCrossTargeting;
-            _configuredProjectContextsByTargetFramework = configuredProjectContextsByTargetFramework;
+            TargetFrameworks = targetFrameworks;
             _configuredProjectsByTargetFramework = configuredProjectsByTargetFramework;
-            _activeTargetFramework = activeTargetFramework;
-            TargetFrameworkProvider = targetFrameworkProvider;
+            ActiveTargetFramework = activeTargetFramework;
+            _targetFrameworkProvider = targetFrameworkProvider;
         }
 
-        private ITargetFrameworkProvider TargetFrameworkProvider { get; }
-
-        public IEnumerable<ITargetedProjectContext> InnerProjectContexts => _configuredProjectContextsByTargetFramework.Values;
-
         public IEnumerable<ConfiguredProject> InnerConfiguredProjects => _configuredProjectsByTargetFramework.Values;
 
-        public ITargetedProjectContext ActiveProjectContext => _configuredProjectContextsByTargetFramework[_activeTargetFramework];
-
-        public bool IsCrossTargeting { get; }
-
-        public void SetProjectFilePathAndDisplayName(string projectFilePath, string displayName)
-        {
-            // Update the project file path and display name for all the inner project contexts.
-            foreach ((ITargetFramework targetFramework, ITargetedProjectContext innerProjectContext) in _configuredProjectContextsByTargetFramework)
-            {
-                // For cross targeting projects, we ensure that the display name is unique per every target framework.
-                innerProjectContext.DisplayName = IsCrossTargeting ? $"{displayName}({targetFramework})" : displayName;
-                innerProjectContext.ProjectFilePath = projectFilePath;
-            }
-        }
-
-        public ITargetedProjectContext GetInnerProjectContext(ProjectConfiguration projectConfiguration, out bool isActiveConfiguration)
+        public ITargetFramework GetProjectFramework(ProjectConfiguration projectConfiguration)
         {
             if (projectConfiguration.IsCrossTargeting())
```

Comment:
A comment explaining what cross targeting is and why returning `TargetFrameworks[0]` is the correct behavior would be helpful.

Reply:
This code will be back on the chopping block before too long. This PR is the result of pulling some provably unused code out in that area, and flattening an unused abstraction. Quite a bit more of it will go later. Getting this PR in now will make later changes easier to review.